### DISCUSSION TITLE
1-line bug (--plot-background: white;)

### DISCUSSION THREAD

**Discussion by ryancwalsh** [observablehq/plot#2081] [Q&A] [answered]
Thanks for your amazing library. We're super excited about it!

This line of this commit was a mistake, right?

![image](https://github.com/observablehq/plot/assets/170662319/20d616d3-e009-4036-b1f7-c89aa3095e27)

https://github.com/observablehq/plot/commit/4baad4dabc22480845afa5fec00447e442b0b13d#diff-30ab29a77cf3887e9e7e3ec68157a17e77e891ca7d6335074ab995ab1f27b82fR258-R265

I know it was meant to address https://github.com/observablehq/plot/issues/1434.

https://github.com/observablehq/plot/blob/17ed0386ccb3187f34affbd0a4dabeff8bf4bcc3/src/plot.js#L261

I assume what we want is:

```css
  background: var(--plot-background);
```

Right now, my tooltips look bad when my app is in dark mode (because the `--plot-background` gets overridden to `white` for this SVG.

![image](https://github.com/observablehq/plot/assets/170662319/23bce47f-2586-4351-b893-535747f897bf)

If you have any quick tips for how to resolve, I'd appreciate it.

Thanks.

**Comment by Fil** [comment, marked as answer]
The corresponding selector is `where(.plot)` which has specificity 0, allowing you to override this color with, e.g.:

```
.plot {
    --plot-background: red;
}
```

or maybe more precisely:
```
.plot [aria-label=tip] {
    --plot-background: red;
}
```

You can wrap this in your dark mode definitions.

**Comment by ryancwalsh** [comment]
Thanks for your fast response!

👉  **I still think this is a legit bug.**

Before you replied, I was able to fix it by forking the `plot` library and fixing the 1 line here:

![image](https://github.com/observablehq/plot/assets/170662319/1cc1e26a-8228-45da-8dab-97552a895582)
https://github.com/walsh-rose-ai/observablehq_plot/commit/e6135a9a5ea7fb543bfa5a734d2c936b539a884d

The result is what I wanted:
![image](https://github.com/observablehq/plot/assets/170662319/66dccfd9-a70d-435f-b125-880923e4f987)

I was able to follow your suggestion to fix it too.

```css
 svg [aria-label=tip] {
     --plot-background: ${themesMap[theme][ThemeVarNames.PlotBackground]};
 }
```

But it feels weird that I'm needing to define `--plot-background` _again_ when I'd already defined it within `:root`.

What do you think?

Thanks.

**Reply by mbostock** [reply]
You can’t define the `--plot-background` property on the `:root`. Setting this property on `:root` has no effect. You need to define it on Plot’s class name as in @Fil’s reply. By default, that is [`.plot-d6a7b5`](https://github.com/observablehq/plot/blob/17ed0386ccb3187f34affbd0a4dabeff8bf4bcc3/src/style.js#L417):

```css
.plot-d6a7b5 {
  --plot-background: darkblue;
}
```

Live example: https://observablehq.com/d/2cc5726c0d0ad080

Furthermore, setting the `--plot-background` variable does not set the background color of the SVG element, which is typically transparent. Instead it makes the desired background color available to marks that need it, such as the tip mark, which needs to fill the tip area to separate its contents visually from the background.

**Reply by ryancwalsh** [reply]
Thank you, @mbostock ! 😃

Very helpful.

**Reply by intcreator** [reply]
have to say this feels very obscure and unintuitive. I don't see it documented on https://observablehq.com/plot/, and it seems weird for this to be set in a CSS file when every other color I've needed to change was set in the Plot API directly
